docs/miscellaneous.rst

```diff
@@ -88,6 +92,264 @@ by checking if the lowest bit is set: short (not set) and long (set).
 .. note::
   Handling invalidly encoded slots is currently not supported but may be added in the future.
 
+JSON Output
+===========
+
+.. _storage-layout-top-level:
+
+The storage layout of a contract can be requested via the :ref:`standard JSON interface <compiler-api>`.  The output is a JSON object containing two keys,
+``storage`` and ``types``.  The ``storage`` object is an array where each
+element has the following form:
+
+
+.. code::
+
+
+    {
+        "astId":"2",
```

Comment:
It is a string, not a number?

Reply:
Updated

Reply:
I'd actually prefer a string here... not every imaginable AST has numbered nodes, so this could actually be any kind of identifier... and I think it doesn't hurt for this output to be as generic as possible - but I'm fine with a number as well...

Reply:
ok!

Reply:
But now the code uses integers.

Reply:
Yes, changed all back to integers except `slot`, and added comment about `slot` in the documentation.